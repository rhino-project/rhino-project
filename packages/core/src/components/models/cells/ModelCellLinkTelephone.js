import { useGlobalOverrides } from 'rhino/hooks/overrides';
import CellLinkTelephone from 'rhino/components/table/cells/CellLinkTelephone';

export const ModelCellLinkTelephoneBase = (props) => (
  <CellLinkTelephone {...props} />
);

const defaultComponents = {
  ModelCellLinkTelephone: ModelCellLinkTelephoneBase
};

const ModelCellLinkTelephone = ({ overrides, ...props }) => {
  const { ModelCellLinkTelephone } = useGlobalOverrides(
    defaultComponents,
    overrides,
    props
  );

  return <ModelCellLinkTelephone {...props} />;
};

export default ModelCellLinkTelephone;
