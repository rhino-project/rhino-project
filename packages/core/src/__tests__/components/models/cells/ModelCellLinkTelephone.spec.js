import { render } from '@testing-library/react';
import { sharedCellTests } from './sharedCellTests';
import ModelCellLinkTelephone from '../../../../../rhino/components/models/cells/ModelCellLinkTelephone';

describe('ModelCellLinkTelephone', () => {
  sharedCellTests(ModelCellLinkTelephone);

  it('renders custom link text', () => {
    const { asFragment } = render(
      <ModelCellLinkTelephone getValue={() => '(231) 232-2334'}>
        Click here to call
      </ModelCellLinkTelephone>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the phone number as link text', () => {
    const { asFragment } = render(
      <ModelCellLinkTelephone getValue={() => '(231) 232-2334'} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
