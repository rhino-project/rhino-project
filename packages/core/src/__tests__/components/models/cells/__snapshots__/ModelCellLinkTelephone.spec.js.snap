// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCellLinkTelephone renders custom link text 1`] = `
<DocumentFragment>
  <a
    href="tel:2312322334"
  >
    Click here to call
  </a>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone renders the phone number as link text 1`] = `
<DocumentFragment>
  <a
    href="tel:2312322334"
  >
    (231) 232-2334
  </a>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render the component with className from inherited props 1`] = `
<DocumentFragment>
  <div
    class="dummy-class"
  >
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellLinkTelephone should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
